### ISSUE TITLE
Loss_D_real and Loss_D_fake remain the same after epoch 1, but Loss_G_L1 keeps decreasing, is it normal?

### ISSUE THREAD

**Issue by lkqnaruto** [junyanz/pytorch-CycleGAN-and-pix2pix#1334]
Hi

I'm adopting pix2pix model to my project. Throughout the training process, I noticed Loss_D_real and Loss_D_fake remain the same after epoch 1, but Loss_G_L1 keeps decreasing. Although it quite weird, but the result is quite decent. I wonder what happened under the hood?

I tried both vanilla GAN loss and wgangp, they both have the similar behavior.

**Comment by junyanz** [comment]
This is pretty normal if you use LSGAN or vanilla GAN loss. They are not supposed to decrease. As long as they are not too small (1e-2), it should be fine. A decreasing Loss_G_L1 is also a good sign.

**Comment by lkqnaruto** [comment]
> This is pretty normal if you use LSGAN or vanilla GAN loss. They are not supposed to decrease. As long as they are not too small (1e-2), it should be fine. A decreasing Loss_G_L1 is also a good sign.

Thank you for the quick reply and your precious time. One follow up question, I also tried wgan, the Loss_D_real and Loss_D_fake have the similar behavior as I mentioned above, so I guess it should be fine?

**Comment by junyanz** [comment]
WGAN loss is not very compatible with PatchGAN, as WGAN assumes that each input sample is independent which is not the case in PatchGAN. We did not have good success with GAN.

**Comment by WayneCho** [comment]
> WGAN loss is not very compatible with PatchGAN, as WGAN assumes that each input sample is independent which is not the case in PatchGAN. We did not have good success with GAN.

In my case, the  the Loss_D_real and Loss_D_fake decrease to a very small value(small than 1e-2) after several steps, dose it means the model is collapsed?

**Comment by lkqnaruto** [comment]
> WGAN loss is not very compatible with PatchGAN, as WGAN assumes that each input sample is independent which is not the case in PatchGAN. We did not have good success with GAN.

Sorry for the late response. Since WGAN loss is not compatible with PatchGAN, do you have any suggestions on what kind of discriminator structure would be a proper candidate? Pixel-discriminator might be a good choice?

**Comment by junyanz** [comment]
If you want to try some new loss, you can try [Hinge loss](https://paperswithcode.com/method/gan-hinge-loss). I don't think there exists a GAN objective loss that can improve results dramatically.

**Comment by lkqnaruto** [comment]
> If you want to try some new loss, you can try [Hinge loss](https://paperswithcode.com/method/gan-hinge-loss). I don't think there exists a GAN objective loss that can improve results dramatically.

Thank you for the recommendation, I'll go ahead and have a try. Appreciate your help.
